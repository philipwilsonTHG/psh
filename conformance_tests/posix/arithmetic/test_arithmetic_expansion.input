# Test arithmetic expansion functionality

# Basic arithmetic operations
echo $((2 + 3))
echo $((10 - 4))
echo $((3 * 4))
echo $((15 / 3))
echo $((17 % 5))

# Test with variables
a=5
b=3
echo $((a + b))
echo $((a * b))

# Test comparison operations
echo $((5 > 3))
echo $((5 < 3))
echo $((5 == 5))
echo $((5 != 3))

# Test logical operations
echo $((1 && 1))
echo $((1 && 0))
echo $((1 || 0))
echo $((0 || 0))

# Test parentheses grouping
echo $(((2 + 3) * 4))
echo $((2 + (3 * 4)))

# Test assignment within arithmetic
echo $((c = 10))
echo $c

# Test increment/decrement
echo $((c++))
echo $c
echo $((++c))
echo $c

# Test with hex and octal
echo $((0xFF))
echo $((010))

# Test bitwise operations
echo $((7 & 3))
echo $((7 | 3))
echo $((7 ^ 3))