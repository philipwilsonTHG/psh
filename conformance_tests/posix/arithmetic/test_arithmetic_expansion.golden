5
6
12
5
2
8
15
1
0
1
1
1
0
1
0
20
14
10
10
10
11
12
12
255
8
3
7
4
